core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -439,6 +439,43 @@ scan (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
 
     extract = extractR . snd
 
+-- | Scan the input of a 'Fold' to change it in a stateful manner using
+-- another 'Fold'. If first fold terminates it will restart again from
+-- initial state.
+-- /Pre-release/
+{-# INLINE scanMany #-}
+scanMany :: Monad m => Fold m a b -> Fold m b c -> Fold m a c
+scanMany (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
+    Fold step initial extract
+
+    where
+
+    {-# INLINE runStep #-}
+    runStep actionL sR = do
+        rL <- actionL
+        case rL of
+            Done bL -> do
+                rR <- stepR sR bL
+                case rR of
+                    Partial sR1 -> runStep initialL sR1
```

Comment:
Only this line is different compared to `scan`. Can we use a common function `scanWith` passing a Bool flag `restart`, and making this line dependent on the bool flag. That way we can reuse the code.

Then:

```
scan = scanWith False
scanMany = scanWith True
```